File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int c=0;
        int l = flowerbed.size();
        for(int i=0;i<l;i++){
            if(n==c){
                return true;
            }
            
            if(flowerbed[i]==0){
                if(i-1>0 && i+1<l){
                    if((flowerbed[i-1]==0) && (flowerbed[i+1]==0)){
                    flowerbed[i]=1;
                    cout << i-1 << " " << i << " " <<i+1 << " " << "a" << endl;
                    c++;
                    if(n==c){
                        return true;
                    }
                    }
                }
                
                else if(i-1<0 && i+1<l){
                    if(flowerbed[i+1]==0){
                       flowerbed[i]=1;
                        cout << i- 1 << " " << i << " " <<i+1 << " " << "b" << endl;
                        c++; 
                        if(n==c){
                            return true;
                        }
                    }
                }
                else if(i-1>=0 && i+1>=l){
                    if(flowerbed[i-1]==0){
                       flowerbed[i]=1;
                        cout << i-1 << " " << i << " " <<i+1 << " " << "c" << endl;
                        c++;
                        if(n==c){
                            return true;
                        }
                    }
                }
              else if(i-1<0 &&i+1>=l){
                  flowerbed[i]=1;
                        cout << i-1 << " " << i << " " <<i+1 << " " << "c" << endl;
                        c++;
                        if(n==c){
                            return true;
                        }
              }
            }
        }
        return false;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int len = flowerbed.length;

        for (int i = 0; i < len; i++) {
            if (flowerbed[i] == 1) continue;

            //Learnt abt Java short-circuit method, if first condition in || is true, it doesn't evaluate the second condition which helps in avoiding the array out of bounds error.
            boolean left = (i == 0 || flowerbed[i - 1] == 0);
            boolean right = (i == len - 1 || flowerbed[i + 1] == 0);

            if (left && right) {
                flowerbed[i] = 1;
                n--;
                if (n == 0) return true;
            }
        }

        return n <= 0;
    }
}
